AWSAppSyncClient/AWSAppSyncSubscriptionWatcher.swift

```diff
@@ -26,30 +26,17 @@ import os.log
 }
 
 private class SubscriptionsOrderHelper {
-    var count = 0
-    var previousCall = Date()
-    var pendingCount = 0
-    var dispatchLock = DispatchQueue(label: "SubscriptionsQueue")
-    var waitDictionary = [0: true]
-    static let sharedInstance = SubscriptionsOrderHelper()
-    
-    func getLatestCount() -> Int {
-        count += 1
-        waitDictionary[count] = false
-        return count
-    }
+    private var count = 0
+    let serialQueue = DispatchQueue(label: "com.amazonaws.appsync.subscriptionsOrderHelper")
     
-    func markDone(id: Int) {
-        waitDictionary[id] = true
-    }
+    static let sharedInstance = SubscriptionsOrderHelper()
     
-    func shouldWait(id: Int) -> Bool {
-        for i in 0..<id where waitDictionary[i] == false {
-            return true
+    var uniqueIdentifier: Int {
```

Comment:
1. Thanks for cleaning up the unused properties
1. Accessing properties shouldn't mutate. Make this a method like `getNextUniqueIdentifier()`
1. It feels a bit heavyweight to use a `DispatchQueue` for synchronizing a counter that you never actually read other than when obtaining the next value. Would a `DispatchSempahore` suffice here?
1. Why not make the counter itself a static member; the instance has no particular use here since it's never overwritten once instantiated
    1. Sub-note: I don't recall that static integers are inherently atomic, so it's probably still necessary to do multi-threading protection, but if you're aware that statics are in fact atomic, then this whole thing pretty much devolves into a static `Int` counter on the `AWSAppSyncSubscriptionWatcher` class.

Reply:
- You are right about the static int not being atomic and thus this approach.
- I agree with having a method and will add one
- I can use a dispatch semaphore, but I chose the DispatchQueue based on my initial reading and seeing what the community is doing. Will dive deep a little more to be sure if I can use DispatchSemaphore
- I'll also consider making the method static

Reply:
`DispatchSemaphore` cannot be used since it would block the calling thread (which could be different in our case)

I am not looking at `DispatchGroup` to validate if it can be used. If not, I might have to fallback on using a queue.

Reply:
Added a test as well to test the behavior.